src/pages/PreexistingCondition.js

```diff
@@ -9,21 +10,80 @@ const PeexistingCondition = ({
   setPage,
   page,
 }) => {
+  const [conditionError, setConditionError] = useState(false);
+  const [isConditionChecked, setIsConditionChecked] = useState(false);
+  const preExistingCondition = {
+    heartDisease: intakeState.heartDisease,
+    cancer: intakeState.cancer,
+    highOrLowBloodPressure: intakeState.highOrLowBloodPressure,
+    diabetes: intakeState.diabetes,
+    asthma: intakeState.asthma,
+    stroke: intakeState.stroke,
+    highCholesterol: intakeState.highCholesterol,
+    rash: intakeState.rash,
+    headacheOrMigrain: intakeState.headacheOrMigrain,
+    depression: intakeState.depression,
+    noPrexistingCondition: intakeState.noPrexistingCondition,
+  };
+
   const handleInputChange = (e) => {
     const item = e.target.name;
     setIntakeState({ ...intakeState, [item]: e.target.value });
   };
 
-  console.log()
-  const handleCheckboxChange = (event) => {
-    const isChecked = event.target.checked;
-    const item = event.target.value;
-    item === "none"
-      ? setIntakeState({ ...intakeState, [item]: !isChecked })
-      : setIntakeState({ ...intakeState, [item]: isChecked });
+  const handleDivSelect = (field) => {
+    setIsConditionChecked(!isConditionChecked);
+    handleOnConditionClick(field);
+  };
+
+  const handleOnConditionClick = (item) => {
+    if (isConditionChecked) {
+      setIsConditionChecked(false);
+    }
+    if (isConditionChecked) {
+      setConditionError(false);
+    }
+
+    if (item === PRE_EXISTING_CONDITION) {
+      setIntakeState({
+        ...intakeState,
+        heartDisease: false,
+        cancer: false,
+        highOrLowBloodPressure: false,
+        diabetes: false,
+        asthma: false,
+        stroke: false,
+        highCholesterol: false,
+        rash: false,
+        headacheOrMigrain: false,
+        depression: false,
+        noPrexistingCondition: true,
+      });
+    } else {
+      setIntakeState({
+        ...intakeState,
+        [item]: isConditionChecked,
+        noPrexistingCondition: false,
+      });
+    }
+  };
+
+  const validateForm = () => {
+    const isAnyTrue = Object.keys(preExistingCondition)
+      .map((key) => preExistingCondition[key])
```

Comment:
.map((key) => preExistingCondition[key])
is redundant here

Reply:
map function removed